### PR TITLE
feat: warning msg on sso login

### PR DESCRIPTION
**Pull request by dhananjay0601**
# Description

Misleading error message while logging in(SSO) if user has no access.

Fixes https://github.com/devtron-labs/devtron/issues/287

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- [X] By logging in SSO

# Checklist:

* [X] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [X] I have performed a self-review of my own code
* [X] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/login/Login.tsx [hunk 1]
```diff
@@ -127,7 +130,14 @@ export default class Login extends Component<LoginProps, LoginFormState> {
 
     renderSSOLoginPage() {
         let search = this.props.location.search
-
+        let renderLoginError = (): JSX.Element => {
```
Comment (sohel-rp) [inline src/components/login/Login.tsx:133]: make it const

File: src/components/login/Login.tsx [hunk 2]
```diff
@@ -148,6 +158,16 @@ export default class Login extends Component<LoginProps, LoginFormState> {
                             </a>
                         )
                     })}
+
+            { !localStorage.isDashboardLoggedIn &&  <InfoColourBar
```
Comment (sohel-rp) [inline src/components/login/Login.tsx:162]: As `localStorage.isDashboardLoggedIn` will return string, check should be something like
```
localStorage.isDashboardLoggedIn !== 'true'
```

File: src/setupProxy.js [hunk 3]
```diff
@@ -19,4 +19,4 @@ module.exports = function (app) {
             secure: false,
         }),
     )
-}
+}//http://demo.devtron.info:32080/
```
Comment (sohel-rp) [inline src/setupProxy.js:22]: remove this file

File: src/config/constantMessaging.ts [hunk 4]
```diff
@@ -57,4 +57,10 @@ export const SSO_NOT_CONFIGURED_STATE_TEXTS = {
         ' Devtron uses Single Sign-On (SSO) to enable one-click login. Please set up an SSO login service before adding users.Go to SSO login services',
     linkText: 'Go to SSO login services',
     redirectLink: '/global-config/login-service',
+}
+
+export const SSO_LOGGING_INFO = {
```
Comment (sohel-rp) [inline src/config/constantMessaging.ts:62]: format

File: src/components/login/Login.tsx (hunk 1 above)
Comment (sohel-rp) [reply src/components/login/Login.tsx:133]: Also format the changes

File: src/components/login/Login.tsx [hunk 5]
```diff
@@ -127,7 +129,19 @@ export default class Login extends Component<LoginProps, LoginFormState> {
 
     renderSSOLoginPage() {
         let search = this.props.location.search
```
Comment (milindk8) [inline src/components/login/Login.tsx:131]: make it const

File: src/components/login/Login.tsx [hunk 6]
```diff
@@ -127,7 +129,19 @@ export default class Login extends Component<LoginProps, LoginFormState> {
...
-
+        const renderLoginError = (): JSX.Element => {
+            return (
+                <>
+                    <span>
+                        {SSO_LOGGING_INFO.frontText}{' '}
```
Comment (milindk8) [inline src/components/login/Login.tsx:136]: remove empty brackects

File: src/components/login/Login.tsx [hunk 7]
```diff
@@ -148,6 +162,17 @@ export default class Login extends Component<LoginProps, LoginFormState> {
                             </a>
                         )
                     })}
+
+                {localStorage.isDashboardLoggedIn !== 'true' && (
+                    <InfoColourBar
+                        classname={'error_bar mt-8 dc__align-left info-colour-bar svg p-8 pl-8-imp mt-20 mb-20 w-300'}
```
Comment (milindk8) [inline src/components/login/Login.tsx:168]: pass classNAme = "xyz" as no string interpolation needed

File: src/css/base.scss [hunk 8]
```diff
@@ -2354,6 +2354,10 @@ textarea,
     padding-left: 16px !important;
 }
 
+.pt-20 {
```
Comment (shivani170) [inline src/css/base.scss:2357]: No need to add this we can directly use pt-20

File: src/setupProxy.js [hunk 9]
```diff
@@ -1,22 +0,0 @@
-const { createProxyMiddleware } = require('http-proxy-middleware')
```
Comment (shivani170) [inline src/setupProxy.js:1]: Revert this please

File: src/components/login/Login.tsx [hunk 10]
```diff
@@ -126,8 +128,20 @@ export default class Login extends Component<LoginProps, LoginFormState> {
     }
 
     renderSSOLoginPage() {
-        let search = this.props.location.search
-
+        const search = this.props.location.search
+        const renderLoginError = (): JSX.Element => {
+            return (
+                <>
+                    <span>
```
Comment (shivani170) [inline src/components/login/Login.tsx:135]: remove this extra span

**Review by milindk8 — APPROVED** [review]

**Pull request merged**